.circleci/src/jobs/build_android.yml

```diff
@@ -0,0 +1,114 @@
+parameters:
+  build_for_delivery:
+    description: Whether builds are delivered to the store. If true, integreat and malte builds are created.
+    type: boolean
+docker:
+  - image: circleci/android:api-28-node
+resource_class: medium
+environment:
+  TOTAL_CPUS: 2 # For resource_class medium, used in metro.config.ci.js.
+  FASTLANE_SKIP_UPDATE_CHECK: true
+shell: /bin/bash -eo pipefail
+steps:
+  - add_ssh_keys: # Needed for credentials repo
+      fingerprints:
+        - "61:d8:94:00:53:5f:3b:19:fd:88:6b:fa:cd:ca:df:5d"
+  - checkout
+  - attach_workspace:
+      at: .
+  - unless: # If not building for delivery, the bump version job has never been executed and therefore no env variables are available
+      condition: << parameters.build_for_delivery >>
+      steps:
+        - run:
+            name: Calculate next version name
+            command: echo "export NEW_VERSION_NAME=$(cat version.json | jq .versionCode)" >> environment_variables
+        - run:
+            name: Calculate next version code
+            command: echo "export NEW_VERSION_CODE=$(cat version.json | jq .versionCode)" >> environment_variables
+  - restore_environment_variables
+  - restore_cache:
+      keys:
+        - 2-gems-{{ arch }}-{{ checksum "native/android/Gemfile.lock" }}
+        - 2-gems-{{ arch }}-
+  - run:
+      command: bundle config set path 'vendor/bundle'
+      working_directory: native/android
+  - run:
+      command: bundle check || bundle install
+      working_directory: native/android
+  - save_cache:
+      key: 2-gems-{{ arch }}-{{ checksum "native/android/Gemfile.lock" }}
+      paths:
+        - native/android/vendor/bundle
+  - restore_yarn_cache
+  - run:
+      name: "[FL] Prepare Android Keystore"
+      command: bundle exec fastlane keystore
+      working_directory: native/android
+  - restore_cache:
+      keys:
+        - 1-gradle-{{ checksum "native/android/build.gradle" }}-{{ checksum "native/android/app/build.gradle" }}-{{ checksum "native/android/settings.gradle" }}
+        - 1-gradle-
```

Comment:
Is seems like something is missing here. Or is there?

Reply:
This is fine. Gradle will use an older cache if the gradle files change.